app/models/marc_indexer.rb

```diff
@@ -63,6 +63,15 @@ def initialize
 
       # configurable logging, quiet by default
       provide 'log.level', ENV['TRAJECT_LOG_LEVEL'] || 'warn'
+      if c = Blacklight.connection_config
+        url = URI.parse(c[:url])
+        if url.user
+          client = HTTPClient.new
+          client.set_auth(c[:url], url.user, url.password)
+          provide "solr_json_writer.http_client", client
+        end
+        provide "solr.url", c[:url]
```

Comment:
I don't know `provide`. how does this work?

Reply:
I don't know, it's a traject config thing.

Reply:
Hey @tpendragon brought this to my attention. This works! But want to make you aware, of:

a) Instead of instantiating your own HTTPClient, you could have provided the auth values in the (somewhat under-documented) `solr_writer.basic_auth_user` and `solr_writer.basic_auth_password` traject config settings.  https://github.com/traject/traject/commit/2404a1f2cd6f3476cc3640d532a6d3a26a3c71a5

b) I plan to fix the SolrJsonWriter you are using in traject here to be able to use basic auth from the URL straight, ie `https://user:pass@host/etc`. https://github.com/traject/traject/issues/261  Should be in a traject release soon. Sorry it wasn't before now!

Reply:
Thanks @jrochkind! We use this instead of the param because the version of traject in this app is older than that feature. We'll keep it in mind when we update though!

Reply:
Oh ok sorry for the unprompted unuseful advice! Upgrading should be very easy if you ever want to, there are very few backwards incompats in 3.0 that should have easy formulaic fixes, and should be no backwards incompats otherwise. But nothing wrong with what you're doing!